pandas/core/generic.py

```diff
@@ -7845,17 +7845,11 @@ def interpolate(
             obj, should_transpose = self, False
         else:
             obj, should_transpose = (self.T, True) if axis == 1 else (self, False)
+            # GH#53631
             if np.any(obj.dtypes == object):
-                # GH#53631
-                if not (obj.ndim == 2 and np.all(obj.dtypes == object)):
-                    # don't warn in cases that already raise
-                    warnings.warn(
-                        f"{type(self).__name__}.interpolate with object dtype is "
-                        "deprecated and will raise in a future version. Call "
-                        "obj.infer_objects(copy=False) before interpolating instead.",
-                        FutureWarning,
-                        stacklevel=find_stack_level(),
-                    )
+                raise TypeError(
```

Comment:
I'd suggest doing this check inside Block.interpolate

Reply:
there's a TODO(3.0) comment related to this inside Block.interpolate

Reply:
thank you, I found the comment. In #58083 I moved raising TypeError from `NDFrame` to `Block`. 

I think maybe it's better to change the error message from  `"NumpyBlock cannot interpolate with object dtype."` to `"Can not interpolate with object dtype."`?